doc/decisions/STEPS.md

```diff
@@ -0,0 +1,141 @@
+
+Short summary:
+
+1. The first PR for a decision is created as "Drafts" (recommended!), "Problem Clear", "Alternatives Clear" or "Decided".
+2. A decision PR that wants to merge a decision as "Decided" must:
+   1. Only contain changes to one decision.
+   2. Already start with the decision as "Decided".
+   3. Already have a clearly stated problem and all the considered alternatives fully explored.
+   4. Only contain discussions/reviews about deciding between one of the considered alternatives or details thereof.
+   5. If there is any change in the direction of the decision then the PR cannot be merged as "Decided". The decision must at least be moved back to "In Review".
+3. After a decision is merged as "Decided", it can be moved to "Partially Implemented" or "Implemented" by any PR.
+
+The rest of the document describes the steps in full details.
+
+## Drafts
+
+> This step is highly recommended and it is even required if the problem is not yet clear to all the core developers.
+
+The first step is to create a PR with:
+
+- **one** decision, where at least the "Problem" is filled out and "Decision", "Rationale" and "Implications" are **not** yet filled out.
+- optional backlinks from related decisions.
+
+This step is brainstorming for completely new ideas.
+
+> No agreement about anything is needed to merge decisions in this stage.
+
+## Problem Clear
+
+This step is very important:
+
+- it clarifies the importance of the problem, answering:
+  Why should we put our time and energy in this problem and not in another problem.
+- it clarifies the scope of the decision.
+- it clarifies relation to other problems.
+
+Decisions will have much smoother further steps if this step is done carefully without prejudice.
+It is especially important that one shouldn't have a fixed mind-set about a preferred solution from the beginning.
+
+> Everyone must agree that the problem exists and is worth solving so that a decision PR in "Problem Clear" step can be merged.
+> A problem is clear if everyone would be able to describe an experiment or test case that shows if a solution fixes the problem.
+
+## Alternatives Clear
+
+> This step is recommended if it is not yet clear to the core developers which solution is the best.
+
+Here you must ensure:
+
+- problem, constraint and assumptions are well-explained and sound
+- consistency with other decisions
+- links from/to related decisions are created
+- there are several considered alternatives, each with rationale and implication
+- "Decision", "Rationale" and "Implications" are **not** yet filled out if there are people arguing for different options
+
+Here the decision should not only have one decision but should describe several solutions.
+For each solution a proposal, rationale and optionally implications should be given.
+
+> The alternatives are clear if all reviewers understand the given alternatives and no reviewers can come up with new alternatives.
```

Comment:
"no reviewers can come up with new alternatives" is a very bad criteria IMO. You can (almost) always come up with more alternatives. It also necessarily leads to long lists of alternatives that were never seriously considered because they are obviously flawed. But somebody had the idea, so we have to list them according to this definition...

Reply:
I agree, f832f42104bef6ada4da94c061cf698d928a19ff

```diff
-> and no reviewers can come up with new alternatives.
+> and no reviewer can come up with better alternatives.
```